### PR TITLE
NEW Make performance guide link configurable

### LINKED ISSUE #58: Make performance guide URL configurable
As per @chillu's comment [here](https://github.com/lekoala/silverstripe-debugbar/pull/40#pullrequestreview-49359141)

### PR DESCRIPTION
**Pull request by robbieaverill**
Resolves #58

### CODE REVIEW

**Comment by robbieaverill** [comment]
Should probably be an enhancement

File: code/DebugBarDatabaseCollector.php [hunk 1]
```diff
@@ -34,12 +34,13 @@ public function collect()
         // Check for excessive number of queries
         $dbQueryWarningLevel = DebugBar::config()->warn_query_limit;
         if ($dbQueryWarningLevel && $data['nb_statements'] > $dbQueryWarningLevel) {
+            $helpLink = DebugBar::config()->performance_guide_link;
```
Comment (Firesphere) [inline code/DebugBarDatabaseCollector.php:37]: `->get()`

**Comment by Firesphere** [comment]
Needs documentation

**Review by sachajudd — APPROVED** [review]

File: _config/debugbar.yml [hunk 2]
```diff
@@ -21,7 +21,9 @@ DebugBar:
   # Maximum number of queries to display
   query_limit: 200
   # Number of DB queries before a warning will be displayed
-  warn_query_limit: 100
+  warn_query_limit: 10
+  # The link to show for large numbers of database queries, and other performance issues
```
Comment (Firesphere) [inline _config/debugbar.yml]: Your description is a bit confusing

I'd say:
"Link to show when there are notable performance issues, e.g. a large number of database queries"

Comment (robbieaverill) [reply _config/debugbar.yml]: There's very little difference in your suggestion, but I've updated the comment to be more clear anyway.

File: code/DebugBarDatabaseCollector.php (hunk 1 above)
Comment (robbieaverill) [reply code/DebugBarDatabaseCollector.php:37]: It's fine the way it is

Comment (Firesphere) [reply code/DebugBarDatabaseCollector.php:37]: How can directly accessing a protected value be "fine"?

Comment (robbieaverill) [reply code/DebugBarDatabaseCollector.php:37]: @Firesphere in 3.x you can access class configuration directly on a Config_ForClass instance or using `get()` - [see here](https://github.com/silverstripe/silverstripe-framework/blob/3/core/Config.php#L916-L918) for where it's implemented

Comment (Firesphere) [reply code/DebugBarDatabaseCollector.php:37]: I'm well aware of the magic getters and setters. I disagree it's good/fine practice to do so. As it's worth a lot to all SilverStripe devs, using best practices over "it works, it's fine" should be applied.

Comment (robbieaverill) [reply code/DebugBarDatabaseCollector.php:37]: Thanks for the feedback. This approach isn't bad practice at all, is valid use of the SS API and is consistent with the rest of the existing module in terms of how config is used.

**Pull request merged**